src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -259,21 +291,25 @@ private void RefreshContextForMoniker(string moniker, IVsHierarchy hierarchy)
             {
                 _foregroundAffinitization.AssertIsForeground();
 
-                _workspace.ApplyChangeToWorkspace(w =>
+                _workspaceApplicationQueue.AddWork(async () =>
```

Comment:
Check if this causes issues with navbar switching contexts for linked files; I could imagine this was previously synchronous, but it's now async and might result in odd behavior -- if you pick a new project from the dropdown, we might not actually update the workspace so the dropdown might not actually refresh until some time later.

Reply:
So trying it out manually seems to work just fine!  i am nervous here though.